### ISSUE TITLE
Different behaviour of AutoTokenizer and T5Tokenizer

### ISSUE THREAD

**Issue by pietrolesci** [huggingface/transformers#19452]
### System Info

- `transformers` version: 4.22.2
- Platform: Linux-5.4.209-129.367.amzn2int.x86_64-x86_64-with-glibc2.26
- Python version: 3.9.13
- Huggingface_hub version: 0.10.0
- PyTorch version (GPU?): 1.12.1+cu102 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: NO
- Using distributed or parallel set-up in script?: NO

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

The `T5Tokenizer` prepends a whitespace before the eos token when a new eos token is provided, while AutoTokenizer maintains the usual behaviour.

```
from transformers import AutoTokenizer, T5Tokenizer

text = ["My name is Pietro", "I love pizza"]

tok = T5Tokenizer.from_pretrained("t5-small", bos_token="[bos]", eos_token="[eos]", sep_token="[sep]")
auto_tok = AutoTokenizer.from_pretrained("t5-small", bos_token="[bos]", eos_token="[eos]", sep_token="[sep]")
print(tok.batch_decode(tok(text)["input_ids"]))
print(auto_tok.batch_decode(tok(text)["input_ids"]))

#> ['My name is Pietro [eos]', 'I love pizza [eos]']
#> ['My name is Pietro[eos]', 'I love pizza[eos]']


tok = T5Tokenizer.from_pretrained("t5-small")
auto_tok = AutoTokenizer.from_pretrained("t5-small")
print(tok.batch_decode(tok(text)["input_ids"]))
print(auto_tok.batch_decode(tok(text)["input_ids"]))

#> ['My name is Pietro</s>', 'I love pizza</s>']
#> ['My name is Pietro</s>', 'I love pizza</s>']
```

### Expected behavior

The two tokenizer classes should be equivalent

**Comment by LysandreJik** [comment]
When you're using `T5Tokenizer`, you're loading the slow version. When using `AutoTokenizer`, you're loading the fast version by default. 

Unfortunately, due to a difference in implementation, the fast and slow tokenizer can have some differences.

Pinging @SaulLu and @ArthurZucker for knowledge

**Comment by ArthurZucker** [comment]
Yes, noticed the same thing with GPT2, especially with OOV that are automatically converted to blank with the fast version!
